controllers/replica_set_controller.go

```diff
@@ -501,7 +502,7 @@ func (r ReplicaSetReconciler) buildAutomationConfig(mdb mdbv1.MongoDBCommunity)
 	}
 
 	auth := automationconfig.Auth{}
-	if err := scram.Enable(&auth, r.client, mdb); err != nil {
+	if err := scram.Enable(&auth, r.client, mdb, []metav1.OwnerReference{getOwnerReference(mdb)}); err != nil {
```

Comment:
I kinda feel we should extract getOwnerReference to somewhere else and we can call it directly whenever we need it instead of passing it by to the Enable function. 

I know it wont't save too much code but it looks cleaner to me

Reply:
+1 to the suggestion.

Reply:
refactored this part after chatting with @bznein : getOwnerReference is not part of MongoDBCommunity and the Configurable interface